app/models/notification_log_reporting.rb

```diff
@@ -0,0 +1,45 @@
+# NOTE: This we need to be updated once a disabled flag is added as it will
+# affect the reporting logic.
+class NotificationLogReporting
+  def initialize(period)
+    @scope = NotificationLog.where(created_at: period)
+  end
+
+  # Generate a list of duplicated gov request ID coming through from a single app
+  # this would be a sign we are sending emails twice.
+  def duplicates
+    @scope
+      .group(:app, :gov_request_id)
+      .having('COUNT(1) > 1')
+      .pluck(:app, :gov_request_id, 'ARRAY_AGG(gov_delivery_ids)')
+  end
+
+  # Report when a notification is sent by GovUkDelivery and not by EmailAlertApi.
```

Comment:
Assuming `gov_delivery_ids` is the topics, isn't this the number of topics that aren't notifying via EmailAlertApi, rather than the number of notifications itself?